fiftyone/utils/torch.py

```diff
@@ -519,8 +587,53 @@ def __init__(self, d):
         self.device = self.parse_string(d, "device", default=None)
 
 
+class ImageGetItem(GetItem):
+    """GetItem for :class:`TorchImageModels`"""
+
+    def __init__(
+        self,
+        transform=None,
+        raw_inputs=False,
+        using_half_precision=False,
+        use_numpy=False,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.transform = transform
+        self.raw_inputs = raw_inputs
+        self.using_half_precision = using_half_precision
+        self.use_numpy = use_numpy
+
+    def __call__(self, d):
+        img = _load_image(
+            # hardcoded because that's how it was in fiftyone.core.models._make_data_loader
+            d["filepath"],
+            use_numpy=self.use_numpy,
+            force_rgb=True,
+        )
+        if self.transform is not None:
+            img = self.transform(img)
+
+        if self.raw_inputs:
+            return img
+
+        else:
+            if self.using_half_precision and torch.is_tensor(img):
+                img = img.half()
+
+        return img
+
+    @property
+    def required_keys(self):
+        return ["filepath"]
+
+
 class TorchImageModel(
-    TorchEmbeddingsMixin, fom.TorchModelMixin, fom.LogitsMixin, fom.Model
+    fom.SupportsGetItem,
```

Comment:
Just to clarify, this one line has **huge** consequences- the vast majority of zoo models are `TorchImageModel`, which means that they are *all* now going to use `to_torch()` to run inference when passed to `apply_model()`, `compute_embeddings()`, etc.

Are we really ready to make that commitment? 😄 

For example, I'm pretty sure upon static review that `required_keys` would not actually work as-is due to `select_fields()`.

Reply:
It's not a matter of my comfort. We just need our zoo models to continue to work. So we need to test all models that now use `SupportsGetItem` to ensure they do.

We have these tests as a starting point for that:
https://github.com/voxel51/fiftyone/blob/develop/tests/intensive/model_zoo_tests.py